crates/bevy_ecs/src/bundle.rs

```diff
@@ -14,120 +14,185 @@ use bevy_ecs_macros::all_tuples;
+/// There is also intentionally no [`Query`] filter which detects whether an entity contains the components
+/// of a bundle.
+/// To select multiple components in a query, you should specify the components exactly.
+///
+/// ## Removal
+///
+/// Bundles can also be used to remove components from an entity.
+///
+/// Removing a bundle from an entity will remove the components of the set it
+/// represents from the entity.
+/// A value of the [`Bundle`] type may then be collected containing these component values,
+/// and returned.
+///
+/// # Implementors
+///
+/// A [`Component`] can be added to or removed from a world.
+/// Therefore, every type which implements [`Component`] also implements [`Bundle`].
+///
+/// Additionally, [Tuples](`tuple`) of bundles are also [`Bundle`] (with up to 15 bundles).
+/// These bundles contain the items of the 'inner' bundles.
+/// This is a convenient shorthand which is primarily used when spawning entities.
+/// For example, spawning an entity using the bundle `(SpriteBundle {...}, PlayerMarker)`
+/// will spawn an entity with components required for a 2d sprite, and the `PlayerMarker` component.
+///
+/// [`unit`], otherwise known as [`()`](`unit`), is a [`Bundle`] containing no components (since it
+/// can also be considered as the empty tuple).
+/// This can be useful for spawning large numbers of empty entities using
+/// [`World::spawn_batch`](crate::world::World::spawn_batch)
+///
+/// Tuple bundles can be nested, which can be used to create an anonymous bundle with more than
+/// 15 items.
+/// However, in most cases where this is required, there is a [`derive@Bundle`] derive instead.
+/// This is used to create your own bundles from a struct of other [`Bundle`]s.
+/// As explained above, this includes any [`Component`] type, and other derived bundles:
 ///
-/// You can nest bundles using the `#[bundle]` attribute:
 /// ```
 /// # use bevy_ecs::{component::Component, bundle::Bundle};
 ///
 /// #[derive(Component)]
-/// struct X(i32);
-/// #[derive(Component)]
-/// struct Y(u64);
+/// struct XPosition(i32);
 /// #[derive(Component)]
-/// struct Z(String);
+/// struct YPosition(i32);
 ///
 /// #[derive(Bundle)]
-/// struct A {
-///     x: X,
-///     y: Y,
+/// struct PositionBundle {
+///     // A bundle can contain components
+///     x: XPosition,
+///     y: YPosition,
 /// }
 ///
 /// #[derive(Bundle)]
-/// struct B {
-///     #[bundle]
-///     a: A,
-///     z: Z,
+/// struct NamedPointBundle {
+///     // Or other bundles
+///     a: PositionBundle,
+///     // In addition to more components
+///     z: PointName,
 /// }
+///
+/// #[derive(Component)]
+/// struct PointName(String);
 /// ```
 ///
 /// # Safety
 ///
+/// Note: Manual implementations of this trait are *not* permitted. That is, there may be arbitrary additional
+/// safety requirements not listed here.
+/// If you want a type to implement [`Bundle`], you must use [`derive@Bundle`].
```

Comment:
```suggestion
/// If you want a type to implement [`Bundle`], you must use the [`derive@Bundle`] derive macro.
```

The link just renders as `Bundle`, so I added some more context.